### PR TITLE
Add ability filter contacts by deleted on API, new index

### PR DESCRIPTION
**Pull request by nicpottier**
Fixes: rapidpro/rapidpro#198

### CODE REVIEW

**Comment by nicpottier** [comment]
plz review @rowanseymour

File: temba/api/tests.py [hunk 1]
```diff
@@ -1456,6 +1474,8 @@ def test_api_contacts(self):
         self.assertIsNotNone(json.loads(response.content)['uuid'])
         self.assertEquals(201, response.status_code)
 
+    test_api_contacts.active = True
```
Comment (rowanseymour) [inline temba/api/tests.py]: Is there a reason for this?

Comment (nicpottier) [inline temba/api/tests.py]: Oops, turd from me testing.

File: temba/api/views.py [hunk 2]
```diff
@@ -1667,7 +1667,13 @@ def destroy(self, request, *args, **kwargs):
             return Response(status=status.HTTP_204_NO_CONTENT)
 
     def get_base_queryset(self, request):
-        return self.model.objects.filter(org=request.user.get_org(), is_active=True, is_test=False)
+        queryset = self.model.objects.filter(org=request.user.get_org(), is_test=False)
+
+        # if they pass in deleted=true then only return deleted contacts
+        if request.QUERY_PARAMS.get('deleted', '').lower() == 'true':
```
Comment (rowanseymour) [inline temba/api/views.py]: Should use `str_to_bool`

**Comment by rowanseymour** [comment]
The alternative solution here would be to have the flag include deleted alongside non-deleted. The advantage there is that you can update and delete modifications after date X in one request. The downside is that the serialized contact would have to include a deleted field.

**Comment by nicpottier** [comment]
Ya, personally I think it would be pretty weird to have the endpoint return deleted contacts by default. Would also be changing our contract on the behavior of the endpoint.

**Comment by rowanseymour** [comment]
It wouldn't be by default - just that the `deleted` parameter would mean "include deleted" rather than "only return deleted"

**Comment by nicpottier** [comment]
Oh I see.. hrmm... ya, that seems weird in that we'd then have to always include ```deleted``` to be consistent in our return format. You ok with this approach?

**Comment by rowanseymour** [comment]
Not following what you mean about consistency. Omitting `deleted` would always exclude deleted contacts.

But I'm fine with this approach.

**Comment by nicpottier** [comment]
Was just saying that the contact JSON would have to include a ```deleted``` field in either case to be consistent. (same thing you pointed out)

**Comment by rowanseymour** [comment]
Then yarp!

**Pull request merged**